### PR TITLE
Add listassetpool, consoleList, fix various problems

### PR DESCRIPTION
**Pull request by iAmThatMichael**
- Fix console display line offset not being reset after calling clear
- Fix opening server menu in a match causing LUI errors to show in console
- Fix LUI crashing whenever calling a map inside a map which was only seen clientside
- Lower the version text display in console to fix text overlapping
- Add listassetpool command replacing the specific pool listing commands
- Add consoleList command to display cmd/dvars

### CODE REVIEW

File: src/client/component/command.cpp [hunk 1]
```diff
@@ -254,6 +254,76 @@ namespace command
...
+					}
+					else
+					{
+						game_console::print(game_console::con_type_info, "[DVAR]\t%s \"%s\"", match.c_str(), game::Dvar_ValueToString(dvar, dvar->current));
+					}
+				}
+				
+				game_console::print(game_console::con_type_info, "Total %i matches", matches.size());
+			});
+
```
Comment (momo5502) [inline src/client/component/command.cpp:326]: What a murder lambda, pls extract that into smaller functions

File: src/client/component/command.cpp [hunk 2]
```diff
@@ -291,6 +361,72 @@ namespace command
...
+								}
+							}
+						}
+						table++;
+					} while (db_size--);
+					game_console::print(game_console::con_type_info,
+									"Total of type: %d | Overrides: %d", 
+										num_type_assets, num_type_override_assets);
+				}
+			});
```
Comment (momo5502) [inline src/client/component/command.cpp:429]: Same here, not gonna review a 200 line function

File: src/client/component/game_console.cpp [hunk 3]
```diff
@@ -542,6 +543,9 @@ namespace game_console
 	{
 		if (key == game::keyNum_t::K_F10)
 		{
+			if (game::mp::svs_clients[localClientNum].header.state >= 1)
+				return false;
```
Comment (momo5502) [inline src/client/component/game_console.cpp:547]: Brackets please:
 if()
 {
 //
}

File: src/client/game/symbols.hpp [hunk 4]
```diff
@@ -41,6 +41,7 @@ namespace game
 
 	WEAK symbol<void(XAssetType type, void (__cdecl* func)(XAssetHeader, void*), void* inData, bool includeOverride)>
 	DB_EnumXAssets_FastFile{0x14017D7C0, 0x14026EC10};
+	WEAK symbol<const char* (const XAsset* asset)> DB_GetXAssetName{ 0x140151C00, 0x140240DD0 };
```
Comment (momo5502) [inline src/client/game/symbols.hpp:44]: No spaces within the brackets:
{0x140151C00, 0x140240DD0}

File: src/client/game/symbols.hpp [hunk 5]
```diff
@@ -193,6 +194,8 @@ namespace game
 	WEAK symbol<SOCKET> query_socket{0, 0x14B5B9180};
 
 	WEAK symbol<void*> DB_XAssetPool{0x140804690, 0x1409B40D0};
+	WEAK symbol<unsigned int> db_hashTable{ 0x142C3E050, 0x143716B10 };
+	WEAK symbol<XAssetEntry> g_assetEntryPool{ 0x142CC2400, 0x14379F100 };
```
Comment (momo5502) [inline src/client/game/symbols.hpp:198]: same here

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/component/command.cpp (hunk 2 above)
Comment (momo5502) [reply src/client/component/command.cpp:382]: this bro.

**Review by momo5502 — CHANGES_REQUESTED** [review]
listassetpool lambda is still a fucking bomb, rest is fine tho

File: src/client/component/command.cpp [hunk 6]
```diff
@@ -226,6 +226,46 @@ namespace command
 		}
 	}
 
+	void list_asset_pool(const game::XAssetType type)
```
Comment (iAmThatMichael) [inline src/client/component/command.cpp:229]: it'd be better if I could move this into fastfiles.cpp would make somewhat more sense there but for now it should be okay in here, commands would require fastfile header which idk if you want to introduce in here

File: src/client/component/command.cpp [hunk 7]
```diff
@@ -226,6 +226,46 @@ namespace command
...
+					while (hash_override)
+					{
+						const auto* entry_override = &game::g_assetEntryPool[hash_override];
+						hash_override = game::g_assetEntryPool[hash_override].nextOverride;
+						game_console::print(game_console::con_type_info, "%s | override",
+											game::DB_GetXAssetName(&entry_override->asset));
+					}
+				}
+			}
+			table++;
```
Comment (momo5502) [inline src/client/component/command.cpp:265]: That's still a bomb, but it's ok

File: src/client/component/command.cpp (hunk 6 above)
Comment (momo5502) [reply src/client/component/command.cpp:229]: Isn't there DB_EnumXAssets which does exactly that?

Comment (iAmThatMichael) [reply src/client/component/command.cpp:229]: DB_EnumXAssets may be a replacement if it doesn't keep track overridden assets then I wouldn't recommend it, the point of listassetpools is to track where assets are loaded from in each respected fastfile eventually I want to add support for display the zone it's loaded from

Comment (momo5502) [reply src/client/component/command.cpp:229]: void __cdecl DB_EnumXAssets_Internal(XAssetType type, void (__cdecl *func)(XAssetHeader, void *), void *inData, _BOOL1 includeOverride)

You can choose if you want them or not. Looks like a better solution to me

Comment (iAmThatMichael) [reply src/client/component/command.cpp:229]: I'll double check it in the morning to see if it's viable, I'm pretty sure I tried messing with it but it required the struct data for each asset type, which we don't have.

Comment (momo5502) [reply src/client/component/command.cpp:229]: I highly doubt it requires that, but yeah, double check it please

Comment (iAmThatMichael) [reply src/client/component/command.cpp:229]: After talking to others it's come to the conclusion we can't get DB_EnumXAssets_Internal to work as the callback parameter is C-style function pointer which doesn't allow a lambda capture which is necessary to know what type is being processed to create the XAsset, this is the only method that's viable for now.

**Review by momo5502 — APPROVED** [review]

**Pull request merged**